### PR TITLE
maps: conditionally display map & warning messages

### PR DESCRIPTION
**Pull request by codemacabre**
## Summary
+ Pass variables `any_nodes_with_geometry` and `any_spans_with_geometry` to template
+ Conditionally show/hide map based on data validity
+ Conditional messages based on data validity
+ Conditionally colour 'Visualisation' section header based on data validity

https://github.com/Open-Telecoms-Data/cove-ofds/issues/6

## Verify
1. Upload valid JSON
   + 'Visualisation' section header will be success green and map should display  
     image

2. Upload JSON without valid geometry
   + 'Visualisation' section header will be warning yellow, along with appropriate message, and map will not display  
     image

3. Upload invalid JSON
   + 'Visualisation' section header will be danger red, along with appropriate message, and map will not display  
     image

### CODE REVIEW

**Review by Lathrisk — COMMENTED** [review]
This looks good to me. The linting changes are quite extensive, but also pass our automated linting checks in the repo. I guess we don't want to regularly change code styles if we use different auto-linters, but otherwise looks good. Any thoughts, @odscjames?

File: cove_ofds/templates/cove_ofds/explore.html [hunk 1]
```diff
@@ -268,21 +268,25 @@ <h4 class="panel-title">
...
+                <p>{% trans 'Your data cannot be visualised on a map because it does not contain coordinates. If you expected your data to include coordinates, you should check your mapping and data pipeline for errors. For more information, see ' %}<a href="https://open-fibre-data-standard.readthedocs.io/en/0.1-dev/reference/schema.html#geometry">{% trans 'Geometry' %}</a>.</p>
+              </div>
+            {% elif can_download_geojson == False %}
               <div>
                 <p>{% trans 'Your data cannot be visualised because it is not available in GeoJSON format. For more information, see ' %}<a href="#download-panel">{% trans 'data conversion' %}</a>.</p>
               </div>
+            {% else %}
+              <p>{% trans 'The GeoJSON version of your data is visualised on the map below. You should check that nodes and spans appear in the correct location. If not, you may need to ' %}<a href="https://open-fibre-data-standard.readthedocs.io/en/0.1-dev/guidance/publication.html#how-to-transform-coordinates-to-the-correct-coordinate-reference-system">{% trans 'transform your coordinates to the correct coordinate reference system' %}</a>.</p>
```
Comment (ghost) [inline cove_ofds/templates/cove_ofds/explore.html]: Noting I'm not sure this way of doing trans tags will work for good translations, but as we don't have I18N at the moment that is fine - all the trans tags will need reviewing anyway

**Comment by ghost** [comment]
We don't have auto-linters for HTML and I don't want to get into that now - ideally PR's would have minimal changes. This allows other PR's to be merged in and preserves the history. Can the 3 commits in this PR be squashed?

Also, noting the messages in the template don't match the 3 in https://github.com/Open-Telecoms-Data/cove-ofds/issues/6#issuecomment-1309319163 . It's not possible to fully do the red message now, as the  {{error details}} template variable isn't available but we could match it as much as possible. That could be in this PR or a new PR after tho.

File: cove_ofds/templates/cove_ofds/explore.html (hunk 1 above)
Comment (ghost) [reply cove_ofds/templates/cove_ofds/explore.html]: feel free to leave trans tags off the messages for now

Comment (codemacabre) [reply cove_ofds/templates/cove_ofds/explore.html]: I followed the existing use of `trans` tags in this file, but happy to adjust as required. Do you want them removed?

Comment (ghost) [reply cove_ofds/templates/cove_ofds/explore.html]: It's fine.

**Review by ghost — APPROVED** [review]

**Pull request merged**